<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="arm_world">
    <plugin filename="gz-sim-physics-system" name="gz::sim::systems::Physics">
      <engine>
        <filename>libgz-physics-bullet-featherstone-plugin.so</filename>
      </engine>
    </plugin>
    <plugin filename="gz-sim-user-commands-system" name="gz::sim::systems::UserCommands"/>
    <plugin filename="gz-sim-scene-broadcaster-system" name="gz::sim::systems::SceneBroadcaster"/>
    <plugin filename="gz-sim-sensors-system" name="gz::sim::systems::Sensors">
      <render_engine>ogre2</render_engine>
    </plugin>
    <gravity>0.0 0.0 -9.8</gravity>
    <scene>
      <shadows>false</shadows>
    </scene>
    <gui>
      <camera name="user_camera">
        <pose>0.75 -0.75 1.4 0 0.29 2.21</pose>
      </camera>
    </gui>
    <include>
      <uri>https://fuel.gazebosim.org/1.0/OpenRobotics/models/Sun</uri>
    </include>
    <model name="table">
      <static>true</static>
      <include>
        <uri>model://table</uri>
      </include>
      <pose>0.29 0 0 0 0 1.5708</pose>
    </model>
    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>1.0</mu>
                <mu2>1.0</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>
    <model name="cube_l_0">
      <pose>0.184 -0.248 1.05 0 0 4.32</pose>
      <link name="cube_l_0_link">
        <collision name="cube_l_0_collision">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
        <visual name="cube_l_0_visual">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
          <material>
            <ambient>0.49 0.87 0.16 1</ambient>
            <diffuse>0.49 0.87 0.16 1</diffuse>
          </material>
        </visual>
        <velocity_decay>
          <linear>1</linear>
          <angular>1</angular>
        </velocity_decay>
      </link>
    </model>
    <model name="cube_l_1">
      <pose>0.466 0.034 1.05 0 0 2.69</pose>
      <link name="cube_l_1_link">
        <collision name="cube_l_1_collision">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
        <visual name="cube_l_1_visual">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
          <material>
            <ambient>0.19 0.79 0.25 1</ambient>
            <diffuse>0.19 0.79 0.25 1</diffuse>
          </material>
        </visual>
        <velocity_decay>
          <linear>1</linear>
          <angular>1</angular>
        </velocity_decay>
      </link>
    </model>
    <model name="cube_l_2">
      <pose>0.286 0.126 1.05 0 0 2.05</pose>
      <link name="cube_l_2_link">
        <collision name="cube_l_2_collision">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
        <visual name="cube_l_2_visual">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
          <material>
            <ambient>0.86 0.68 0.16 1</ambient>
            <diffuse>0.86 0.68 0.16 1</diffuse>
          </material>
        </visual>
        <velocity_decay>
          <linear>1</linear>
          <angular>1</angular>
        </velocity_decay>
      </link>
    </model>
    <model name="cube_m_3">
      <pose>0.446 -0.257 1.05 0 0 4.12</pose>
      <link name="cube_m_3_link">
        <collision name="cube_m_3_collision">
          <geometry>
            <box>
              <size>0.05 0.05 0.05</size>
            </box>
          </geometry>
        </collision>
        <visual name="cube_m_3_visual">
          <geometry>
            <box>
              <size>0.05 0.05 0.05</size>
            </box>
          </geometry>
          <material>
            <ambient>0.15 0.36 0.74 1</ambient>
            <diffuse>0.15 0.36 0.74 1</diffuse>
          </material>
        </visual>
        <velocity_decay>
          <linear>1</linear>
          <angular>1</angular>
        </velocity_decay>
      </link>
    </model>
    <model name="sphere_m_4">
      <pose>0.48 -0.216 1.05 0 0 0.3</pose>
      <link name="sphere_m_4_link">
        <collision name="sphere_m_4_collision">
          <geometry>
            <sphere>
              <radius>0.025</radius>
            </sphere>
          </geometry>
        </collision>
        <visual name="sphere_m_4_visual">
          <geometry>
            <sphere>
              <radius>0.025</radius>
            </sphere>
          </geometry>
          <material>
            <ambient>0.97 0.49 0.18 1</ambient>
            <diffuse>0.97 0.49 0.18 1</diffuse>
          </material>
        </visual>
        <velocity_decay>
          <linear>1</linear>
          <angular>1</angular>
        </velocity_decay>
      </link>
    </model>
  </world>
</sdf>
